pkg/asset/manifests/azure/cluster.go

```diff
@@ -86,6 +91,13 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 					PrivateDNSZoneName: installConfig.Config.ClusterDomain(),
 				},
 				Vnet: capz.VnetSpec{
+					ResourceGroup: installConfig.Config.Azure.NetworkResourceGroupName,
+					Name:          installConfig.Config.Azure.VirtualNetwork,
+					// The ID is set to virtual network here for existing vnets here. This is to force CAPZ to consider this resource as
+					// "not managed" which would prevent the creation of an additional nsg and route table in the network resource group.
+					// The ID field is not used for any other purpose in CAPZ except to set the "managed" status.
+					// See https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/scope/cluster.go#L585
+					// https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/0f321e4089a3f4dc37f8420bf2ef6762c398c400
```

Comment:
Shouldn't this only be set when installing into existing vnets? If this is always set, capz will always mark it as managed, right?

Reply:
If VirtualNetwork isn't set in the install config, the ID is set to "" which is the same as not setting it.

Reply:
Makes sense. I was under the impression it was not empty.